src/utils/common.ts

```diff
@@ -1,26 +1,34 @@
+import { isRouteErrorResponse } from '@remix-run/react';
+import { isEcomSDKError } from '~/api/types';
+
 export function getUrlOriginWithPath(url: string) {
     const { origin, pathname } = new URL(url);
     return new URL(pathname, origin).toString();
 }
 
-export function toError(value: unknown): Error {
-    if (value instanceof Error) {
-        return value;
+/**
+ * Try to find some error message in unknown value.
+ *
+ * Handles cases when value is `ErrorResponse`, `EcomSDKError`, `String` or `Object`.
+ *
+ * Falls back to 'Unknown error' message.
```

Comment:
This prevents the caller from doing `getErrorMessage(error) || 'Oops'`

Reply:
BTW, any error that doesn't have a message is a bug that needs to fixed rather than worked around with 'Unknown error'.

Reply:
changed return type to string | undefined